src/videojs-media-sources.js

```diff
@@ -191,9 +191,18 @@ addTextTrackData = function (sourceHandler, captionArray, metadataArray) {
       // MediaSource
       this.sourceBuffers = [];
 
+      // Re-emit MediaSource events on the polyfill
+      [
+        'sourceopen',
+        'sourceclose',
+        'sourceended'
+      ].forEach(function(eventName) {
+        this.mediaSource_.addEventListener(eventName, this.trigger.bind(this));
```

Comment:
@dmlap We were re-emitting `sourceopen` because we depended on it in videojs-contrib-hls. The others just fell through the cracks is my guess.